src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1202,6 +1209,18 @@ class DocBaseViewer extends BaseViewer {
                 this.resize();
             }
         }
+
+        // Fire rendered metric to indicate that the specific page of content the user requested has been shown
+        if (!this.startPageRendered && (this.startPageNum === pageNumber || this.getCachedPage() === pageNumber)) {
+            const pageRenderTag = Timer.createTag(this.options.file.id, RENDER_METRIC);
+            const pageRenderTimer = Timer.stop(pageRenderTag) || {};
+
+            this.emitMetric({
+                name: RENDER_EVENT,
+                data: pageRenderTimer.elapsed,
+            });
+            this.startPageRendered = true;
```

Comment:
Should this be declared at the top as a property of the class?

Reply:
We don't do that for any of the other instance properties in this class, save one.